docs/advanced-recipes/atom-creators.mdx

```diff
@@ -291,3 +291,61 @@ function EvenCounter() {
   return <>Count was set to an even number {evenCount} times.</>
 }
 ```
+
+## atomWithBroadcast
+
+> `atomWithBroadcast` creates an atom. The atom will be shared between
+> browser tabs and frames, similar to `atomWithStorage` but with the 
+> initialization limitation.
+
+This can be useful when you want the state to interact with each other without
+the use of localStorage and that by just using The Broadcast Channel API allows
+basic communication between browsing contexts (that is, windows, tabs, frames, 
+create a component or iframes) and workers on the same origin.
+
+```ts
+import { atom } from "jotai";
+
+export function atomWithBroadcast<Value>(key: string, initialValue: Value) {
+  const channel = new BroadcastChannel(key);
+
+
+  const baseAtom = atom(initialValue);
+
+  const broadcastAtom = atom<Value, { fromOnMessage: boolean; value: Value }>(
+    (get) => get(baseAtom),
+    (get, set, update) => {
+      if (!update.fromOnMessage) {
+        channel.postMessage(update.value);
+      }
+      set(baseAtom, update.value);
+    }
+  );
+  broadcastAtom.onMount = (setAtom) => {
+    channel.onmessage = (event) => {
+      setAtom({ fromOnMessage: true, value: event.data });
+    };
```

Comment:
we should clean up something. or is it garbage collectable?

Reply:
We would need to pair onMount and onUnmount.
At least, you should only `.close()` when `listerns` is empty.
You can try having two atoms in the example and one of them behind show/hide toggle.

I'm not very familiar with BroadcastChannel. Can we create multiple channels in a single thread?
```js
const c1 = new BroadcastChannel('foo');
const c2 = new BroadcastChannel('foo');
```

If it's too hard to solve, we could get `channel` from function arguments.

Reply:
solved it in the new commit.
tell me what do u think?

Reply:
Oh, I missed that you added `channel.close` in clean up. If we go with this, we need to create `channel` in onMount. Otherwise, it will work unexpectedly. Basically, onMount and cleanup should be paired. Same applies to `useEffect`.